src/main/java/frc/robot/subsystems/SuperStructure.java

```diff
@@ -168,6 +181,85 @@ public Command coralLevelOne(BooleanSupplier score) {
         .withName("Coral Level 1");
   }
 
+  // New versions with ipScore
+  // if robot is in position for intermediate scoring, it can score early but if vision is dead
+  // manual control still works
+  // only used on solo controls
+
+  public Command coralLevelThree(BooleanSupplier score, BooleanSupplier ipScore) { // same as L4
+    return Commands.sequence(
+            Commands.parallel(
+                    elevator
+                        .setLevel(ElevatorSubsystem.CORAL_LEVEL_THREE_PRE_POS)
+                        .deadlineFor(
+                            armPivot
+                                .moveToPosition(ArmPivot.CORAL_PRESET_UP)
+                                .until(ipScore)
+                                .until(score)),
+                    spinnyClaw.stop())
+                .withTimeout(0.5),
+            repeatPrescoreScoreSwing(
+                Commands.repeatingSequence(
+                    armPivot
+                        .moveToPosition(ArmPivot.CORAL_PRESET_L3)
+                        .withDeadline(Commands.waitUntil(ipScore).until(score)),
+                    armPivot
+                        .moveToPosition(ArmPivot.CORAL_PRESET_DOWN)
+                        .withTimeout(1.5)
+                        .until(armPivot.atAngle(ArmPivot.CORAL_POST_SCORE))),
```

Comment:
There is no explicit error handling in case the subsystems fail to reach their target positions. Consider adding fallback logic or logging to handle such cases. Once physical simulation of the subsystems are implemented, tweaking parameters there (to make them move slower or faster) would allow testing of the error handling here.